rcl_yaml_param_parser/src/impl/yaml_descriptor.h

```diff
@@ -0,0 +1,51 @@
+// Copyright 2018 Apex.AI, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IMPL__YAML_DESCRIPTOR_H_
+#define IMPL__YAML_DESCRIPTOR_H_
+
+#include "rcutils/allocator.h"
+
+#include "./types.h"
+#include "rcl_yaml_param_parser/types.h"
+#include "rcl_yaml_param_parser/visibility_control.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+///
+/// Finalize an rcl_param_descriptor_t
+///
+RCL_YAML_PARAM_PARSER_PUBLIC
+void rcl_yaml_descriptor_fini(
```

Comment:
@bpwilcox did you mean `rcl_param_descriptor_fini`? Same elsewhere. YAML is implied.

Reply:
This parallels the function naming in `yaml_variant.h` so I just kept it. If we change the name to your suggestion, I'd recommend we do the same for the variant version. I don't know if this is a big concern, however.